flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -417,7 +420,12 @@ protected StreamTask(
             this.asyncExceptionHandler = new StreamTaskAsyncExceptionHandler(environment);
 
             this.asyncOperationsThreadPool =
-                    Executors.newCachedThreadPool(
+                    new ThreadPoolExecutor(
+                            0,
+                            configuration.getMaxConcurrentCheckpoints() + 1,
```

Comment:
nit: maybe add a comment explaining this value?
> With maxConcurrentCheckpoints + 1 we more or less adhere to the maxConcurrentCheckpoints configuration, but allow for a small leeway with allowing for simultaneous N ongoing concurrent checkpoints and for example clean up of one aborted one.

Reply:
Thanks a lot for your help, the code has been resubmitted. There are some unknown errors in CI, which don't seem to be related to my modification, do I need to fix it?

Reply:
@pnowojski Thanks for your help, I've resolved the comments and the CI has passed.